dix/req.c

```diff
@@ -1771,35 +1772,57 @@ static int dix_cas_rops_send(struct m0_dix_req *req)
 		cctg_id.ci_layout.dl_type = layout->dl_type;
 		rc = m0_dix_ldesc_copy(&cctg_id.ci_layout.u.dl_desc,
 				       &layout->u.dl_desc);
-		M0_LOG(M0_DEBUG, "Processing dix_req %p[%u] "FID_F
-				  " creq=%p "FID_F,
-				  req, req->dr_type,
-				  FID_P(&req->dr_indices[0].dd_fid),
-				  creq, FID_P(&cctg_id.ci_fid));
+		if (rc == 0) {
```

Comment:
m0_cas_id_fini() is called even if m0_dix_ldesc_copy() failed. This does not look right, cctg_id might contains some garbage fields at this point.

Reply:
Fixed.